presto-hive/src/main/java/com/facebook/presto/hive/InternalHiveSplit.java

```diff
@@ -93,6 +100,20 @@ public InternalHiveSplit(
         // increment the references to partitionInfo.
         // This will be decremented in HiveSplitSource when the split is done.
         partitionInfo.incrementReferences();
+
+        ImmutableList.Builder<List<HostAddress>> addressBuilder = ImmutableList.builder();
+        ImmutableList.Builder<Long> endsBuilder = ImmutableList.builder();
```

Comment:
hmm... why not just use `ArrayList` or `IntArrayList` ? 

For variable name, what about `blockEndOffsetsBuilder` ?